File: src/features/typing/data/go/search_a_2d_matrix.go
package main

func searchMatrix(matrix [][]int, target int) bool {
	m, n := len(matrix), len(matrix[0])
	left, right := 0, m*n
	for left < right {
		mid := left + (right-left)/2
		val := matrix[mid/n][mid%n]
		if val < target {
			left = mid + 1
		} else {
			right = mid
		}
	}
	return left < m*n && matrix[left/n][left%n] == target
}

File: src/features/typing/data/ts/search_a_2d_matrix.ts
function searchMatrix(matrix: number[][], target: number): boolean {
    const m = matrix.length;
    const n = matrix[0].length;
    let left = 0;
    let right = m * n;
    while (left < right) {
        const mid = left + Math.floor((right - left) / 2);
        if (matrix[Math.floor(mid / n)][mid % n] < target) {
            left = mid + 1;
        } else {
            right = mid;
        }
    }
    return left < m * n && matrix[Math.floor(left / n)][left % n] === target;
}
